src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionsController.kt

```diff
@@ -100,16 +100,22 @@ class IdentificationSessionsController(
     }
 
     @GetMapping("/{eIDSessionId}")
-    fun getIdentity(@PathVariable eIDSessionId: UUID): Mono<ResponseEntity<GetResultResponseType>> {
+    fun getIdentity(@PathVariable eIDSessionId: UUID, authentication: Authentication): Mono<ResponseEntity<GetResultResponseType>> {
         /*
             Wrapping blocking call to the SDK into Mono.fromCallable
             https://projectreactor.io/docs/core/release/reference/index.html#faq.wrap-blocking
         */
+        val apiKeyDetails = authentication.details as ApiKeyDetails
         val getIdentityResult = Mono.fromCallable {
             val eidService = EidService(config)
             eidService.getEidInformation(eIDSessionId.toString())
         }
         return identificationSessionService.findByEIDSessionId(eIDSessionId)
+            .doOnNext {
+                if (apiKeyDetails.refreshAddress != it.refreshAddress) {
+                    throw SecurityException("APIKey unauthorized")
```

Comment:
The `SecurityException` docs say "Thrown by the security manager to indicate a security violation." which is not really the case here. Hence, I would recommend to use Springs `AuthenticationException` which is "Abstract superclass for all exceptions related to an Authentication object being invalid for whatever reason." or use an appropriate subclass. Furthermore, I would suggest to add some more info to the exception message since the message will not be externalized but might help us when debugging or understanding the code.

```suggestion
                    throw AuthenticationException("API key differs from the API key used to start the identification session.")
```

Reply:
Yes. Been looking into the [AuthenticationException](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/AuthenticationException.html ) as well, but it is an abstract class and we cannot throw that with an individual message. 
So I figured that the [SessionAutheticationException](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/web/authentication/session/SessionAuthenticationException.html) can be instantiated and looks good for that use case.